modules/gdscript/doc_classes/@GDScript.xml

```diff
@@ -1383,7 +1386,7 @@
 				# All functions are done!
 				[/codeblock]
 				When yielding on a function, the [code]completed[/code] signal will be emitted automatically when the function returns. It can, therefore, be used as the [code]signal[/code] parameter of the [code]yield[/code] method to resume.
-				If you are planning on calling the same function within a loop, you should consider using [code]yield(get_tree(), "idle_frame")[/code] also.
+				In order to yield on a function, the resulting function also needs to inherit from [code]GDScriptFunctionState[/code]. Notice [code]yield(get_tree(), "idle_frame")[/code] from the above example.
```

Comment:
"Inherit from `GDScriptFunctionState`"? You probably meant to say that the function should "return `GDScriptFunctionState`"?

Reply:
> 
> 
> > I'm just not seeing any good suggestion so far.
> 
> Perhaps this example is better:
> 
> ```gdscript
> func _ready():
>     yield(countdown(), "completed") # waiting for the countdown() function to complete
>     print('Ready')
> 
> func countdown():
>     yield(get_tree(), "idle_frame") # returns a GDSCriptFunctionState object to _ready()
>     print(3)
>     yield(get_tree().create_timer(1.0), "timeout")
>     print(2)
>     yield(get_tree().create_timer(1.0), "timeout")
>     print(1)
>     yield(get_tree().create_timer(1.0), "timeout")
> ```

Yes, looks much better to me. and certainly easier to understand.